frontend/src/api/open-hands.ts

```diff
@@ -86,9 +131,12 @@ class OpenHands {
    * @returns Blob of the workspace zip
    */
   static async getWorkspaceZip(conversationId: string): Promise<Blob> {
-    const url = `/api/conversations/${conversationId}/zip-directory`;
+    const baseUrl =
+      this.conversationUrl || `/api/conversations/${conversationId}`;
```

Comment:
We could use the `getConversationUrl` function from `frontend/src/api/file-service/file-service.utils.ts` to replace all the existing `baseUrl` declarations:

```suggestion
    const baseUrl = getConversationUrl(conversationId);
```